contracts/ERC20MultiDelegate.sol

```diff
@@ -0,0 +1,251 @@
+
+        for (uint index = 0; index < delegateePairsLength; index++) {
+            address from = delegateePairs[index].source;
+            address to = delegateePairs[index].target;
+
+            uint256 amount = getBalanceForDelegatee(from);
+            sourceIds[index] = uint256(uint160(from));
+            targetIds[index] = uint256(uint160(to));
+            amounts[index] = amount;
+
+            transferBetweenDelegators(from, to, amount);
+
+            deployProxyDelegatorIfNeeded(to);
+        }
+
+        burnBatch(msg.sender, sourceIds, amounts);
+        mintBatch(msg.sender, targetIds, amounts);
+    }
+
+    /**
+     * @dev Withdraws delegated ERC20 voting power from proxy delegators to the actual delegator
+     * @param delegatees List of delegatee addresses
+     */
+    function withdrawMulti(address[] calldata delegatees) external {
+        uint256 delegateesLength = delegatees.length;
+
+        require(
+            delegateesLength > 0,
+            "WithdrawMulti: You should pick at least one delegatee"
+        );
+
+        uint256[] memory delegates = new uint256[](delegateesLength);
+        uint256[] memory amounts = new uint256[](delegateesLength);
+
+        for (uint256 index = 0; index < delegateesLength; index++) {
+            address delegatee = delegatees[index];
+            uint256 amount = getBalanceForDelegatee(delegatee);
+            delegates[index] = uint256(uint160(delegatee));
+            amounts[index] = amount;
+
+            address proxyAddress = getProxyContractAddress(delegatee);
+            transferVotingPower(proxyAddress, msg.sender, amount);
+        }
+
+        burnBatch(msg.sender, delegates, amounts);
+    }
+
+    function setUri(string memory uri) external onlyOwner {
+        _setURI(uri);
+    }
+
+    function createProxyDelegatorAndTransfer(
+        address delegatee,
+        uint256 amount
+    ) internal {
+        address proxyAddress = deployProxyDelegatorIfNeeded(delegatee);
+        token.transferFrom(msg.sender, proxyAddress, amount);
+    }
+
+    function transferBetweenDelegators(
+        address from,
+        address to,
+        uint256 amount
+    ) internal {
+        address proxyAddressFrom = getProxyContractAddress(from);
+        address proxyAddressTo = getProxyContractAddress(to);
+        token.transferFrom(proxyAddressFrom, proxyAddressTo, amount);
+    }
+
+    function deployProxyDelegatorIfNeeded(
+        address delegatee
+    ) internal returns (address) {
+        (address proxyAddress, bytes32 salt) = retrieveProxyContractAddress(
+            token,
+            delegatee
+        );
+        new ERC20ProxyDelegator{salt: salt}(token, delegatee);
+        return proxyAddress;
+    }
+
+    function getBalanceForDelegatee(
+        address delegatee
+    ) internal view returns (uint256) {
+        return ERC1155(this).balanceOf(msg.sender, uint256(uint160(delegatee)));
+    }
+
+    function getProxyContractAddress(
```

Comment:
Why do we have this function wrapping the othe rone?

Reply:
oh, leftover. going to update it.